<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>about me - nebfield</title>

    <link rel="stylesheet" href="css/style.css">
    <script async src="./dist/browser.js"> </script>
    <script async src="./js/cursor.js"> </script>
</head>

<body>
    <nav>
        <ul class="main-nav">
            <ul><a href="index.html">home</a></ul>
            <ul><a href="about.html">about</a></ul>
            <ul><a href="projects.html">projects</a></ul>
            <ul><a href="science.html">science</a></ul>
        </ul>
    </nav>

    <h1>Projects</h1>

    <h2>Bioinformatics workflows and software</h2>

    <p>I like to make useful, reliable, and correct scientific tools. </p>
    <p>Most of the things I develop <a href="https://github.com/nebfield">are available to view, download, or modify from open source repositories.</a></p>
    <p>Below is a summary of some things I&#39;ve helped to make with various excellent people.</p>

    <h3>Polygenic Score (PGS) Catalog Calculator</h3>
    <p><code><a href="https://github.com/PGScatalog/pgsc_calc/">pgsc_calc</a></code> is a <a
            href="https://nextflow.io/">nextflow</a>
        workflow for Polygenic Score (PGS) calculation. The basic idea is: </p>
    <ul>
        <li>Scientists <a href="https://www.pgscatalog.org/">make and publish</a> polygenic scores using massive amounts
            of genomes and linked healthcare records <ul>
                <li>Polygenic scores predict genetic predisposition for a trait</li>
            </ul>
        </li>
        <li>Genomes and healthcare records are very sensitive data and should never be freely shared</li>
        <li>Polygenic scores only contain a list of variants and weights, so can be freely shared, like <a
                href="https://www.ebi.ac.uk/gwas/">GWAS summary statistics</a></li>
        <li>Other scientists might want to re-use these scores for their own research and cohorts </li>
        <li>It&#39;s difficult to do this correctly without a deep understanding of human genetics and associated tools
        </li>
    </ul>
    <p>The aim of <code>pgsc_calc</code> is to make it easy to re-use polygenic scores. I helped to take research code
        from
        Cambridge University and transform it into a portable, reliable, and scalable workflow to calculate PGS. We
        worked
        hard to make sure scores were calculated correctly across different genetic ancestry groups.</p>
    <h3>snpQT</h3>
    <p><code><a href="https://github.com/nebfield/snpQT">snpQT</a></code> is a nextflow workflow to simplify working
        with human
        genetics data and doing:</p>
    <ul>
        <li>quality control </li>
        <li>population stratification</li>
        <li>genome wide association studies</li>
        <li>imputation </li>
    </ul>
    <p>Sequencing human genomes is quite cheap, so a lot more scientists have access to human genetic data but don&#39;t
        really know how to work with it (myself included at the time). The aim of the workflow was to <a
            href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8637247/">formalise a set of best practices at our
            research centre.</a> We published the workflow hoping others might find it helpful. </p>

    <br />

    <h2>Software development</h2>
    <p>I'm currently working on integrating <code>pgsc_calc</code> with a genetic data analysis platform for the <a
            href="https://www.interveneproject.eu/">INTERVENE project.</a></p>
    <p>This involves building tools to help deploy and monitor the workflow across a diverse set of systems, including:
    </p>
    <ul>
        <li>Kubernetes (on a private cloud)</li>
        <li>SLURM (on a high performance computing cluster)</li>
        <li>Cloud platforms (on public cloud)</li>
    </ul>

    <br />

    <h2>Data science</h2>
    <h3>COVID-19</h3>
    <p>At the start of the pandemic I was part of a small team that helped the <a
            href="https://www.publichealth.hscni.net/">Public Health Agency</a> set up data science workflows and
        applications <a
            href="https://www.ulster.ac.uk/news/2020/july/ulster-researchers-support-the-response-to-covid-19">for
            contact tracing</a>. Public health experts at the PHA guided the development of
        interactive data visualisations, and they were happy to upgrade from <a
            href="https://www.bbc.co.uk/news/technology-54423988">dodgy spreadsheets ðŸ‘€</a></p>

    <p>Unfortunately the developed applications aren't publicly available.</p>

    <br />

    <footer>
        <hr>
        <div class="footer-content">
            <div class="haiku">
                <h3>haiku</h3>
                <p><em>
                        dna is cool<br>
                        often a bit confusing<br>
                        but life always is
                    </em></p>
            </div>
            <div class="cc-license">
                <a rel="license" href="http://creativecommons.org/publicdomain/zero/1.0/">
                    <img src="img/cc0.png" alt="CC0" />
                </a>
            </div>
        </div>
    </footer>
</body>

</html>